handler/category-dimension-import.go

```diff
@@ -186,6 +143,93 @@ func (h *CategoryDimensionImport) Handle(ctx context.Context, e *event.CategoryD
 	return nil
 }
 
+// BatchPatchInstance sends new dimension options to Dataset API, corresponding to the provided Cantabular variable, in batches of up to BatchSizeLimit
+func (h *CategoryDimensionImport) BatchPatchInstance(ctx context.Context, e *event.CategoryDimensionImport, variable cantabular.Variable, eTag string) (newETag string, err error) {
+	// Get batch splits for provided items
+	numFullChunks := variable.Len / h.cfg.BatchSizeLimit
+	remainingSize := variable.Len % h.cfg.BatchSizeLimit
+
+	// process full batches
+	for i := 0; i < numFullChunks; i++ {
+		batchOffset := i * h.cfg.BatchSizeLimit
+		optionsBatch := make([]*dataset.OptionPost, h.cfg.BatchSizeLimit)
+		for j := 0; j < h.cfg.BatchSizeLimit; j++ {
+			optionsBatch[j] = &dataset.OptionPost{
+				Name:     variable.Name,
+				CodeList: variable.Name,                 // TODO can we assume this?
+				Code:     variable.Codes[batchOffset+j], // TODO can we assume this?
+				Option:   variable.Codes[batchOffset+j],
+				Label:    variable.Labels[batchOffset+j],
+			}
+		}
+		eTag, err = h.PatchInstanceDimensionsWithRetires(ctx, e, optionsBatch, eTag, 0)
+		if err != nil {
+			return "", err
```

Comment:
Here and line 186 errors needing context?

Reply:
I created a nested func instead, it will remove duplication and I think it will make the code clearer.